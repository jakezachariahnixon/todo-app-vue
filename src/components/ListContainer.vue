<template>
  <div id="listContainer">
    <h1 id="listName">{{ title }}</h1>
  </div>
</template>

<script>
export default {
  name: "ListContainer",
  props: {
    title: String
  },
  data() {
    return {
      todos: []
    };
  },
  mounted() {
    if (localStorage.getItem("todos"))
      this.todos = JSON.parse(localStorage.getItem("todos"));
  },
  watch: {
    todos: {
      handler() {
        localStorage.setItem("todos", JSON.stringify(this.todos));
      },
      deep: true
    }
  }
};
</script>

<style scoped lang="scss">
@import "../scss/color.scss";
// Container positioning and style
#listContainer {
  position: fixed;
  width: 25%;
  height: 60%;
  top: 20%;
  left: 37.5%;
  background: $white 0% 0% no-repeat padding-box;
  border-radius: 10px;
}
// Add fade out at bottom of the tasks list
#listContainer:after {
  content: "";
  position: absolute;
  bottom: 0;
  left: 0;
  pointer-events: none;
  background-image: linear-gradient(
    to bottom,
    rgba(255, 255, 255, 0),
    rgba(255, 255, 255, 1) 90%
  );
  width: 100%;
  height: 4em;
  border-radius: 10px;
  z-index: 1;
}
// Style list name
#listName {
  padding-top: 4%;
  padding-bottom: 4%;
  text-align: center;
  font-family: Proxima Nova;
  font-weight: Bold;
  font-size: 2vw;
  height: 5%;
  letter-spacing: 0px;
  color: $titleColor;
}
</style>
