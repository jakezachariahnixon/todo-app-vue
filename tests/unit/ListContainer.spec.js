import { mount, shallowMount } from '@vue/test-utils'
import ListContainer from '@/components/ListContainer.vue';


const title = "Example List Title";
const noPropWrapper = mount(ListContainer);
const titledWrapper = mount(ListContainer, {
    propsData: {
        title: "Example List Title"
    }
});
const taskedWrapper = mount(ListContainer, {
    data () {
        return {
            todos: [{"title":"Root Task One","completed":false,"priority":"mid","parent":"","toggle":false},{"title":"Root Task Two","completed":false,"priority":"high","parent":"","toggle":false},{"title":"Root Task Three","completed":false,"priority":"low","parent":"","toggle":false},{"title":"Sub-task One","completed":true,"priority":"low","parent":"Root Task One","toggle":false},{"title":"Sub-task Two","completed":false,"priority":"low","parent":"Root Task One","toggle":false},{"title":"Root Task Four","completed":false,"priority":"mid","parent":"","toggle":false},{"title":"Root Task Five","completed":false,"priority":"low","parent":"","toggle":false},{"title":"Root Task Six","completed":false,"priority":"high","parent":"","toggle":false},{"title":"Sub-task Three","completed":false,"priority":"mid","parent":"Root Task Three","toggle":false}],
        }
    }
});
const completeWrapperInvalidTaskName = mount(ListContainer, {
    data () {
        return {
            todos: [{"title":"Root Task One","completed":false,"priority":"mid","parent":"","toggle":false},{"title":"Root Task Two","completed":false,"priority":"high","parent":"","toggle":false},{"title":"Root Task Three","completed":false,"priority":"low","parent":"","toggle":false},{"title":"Sub-task One","completed":true,"priority":"low","parent":"Root Task One","toggle":false},{"title":"Sub-task Two","completed":false,"priority":"low","parent":"Root Task One","toggle":false},{"title":"Root Task Four","completed":false,"priority":"mid","parent":"","toggle":false},{"title":"Root Task Five","completed":false,"priority":"low","parent":"","toggle":false},{"title":"Root Task Six","completed":false,"priority":"high","parent":"","toggle":false},{"title":"Sub-task Three","completed":false,"priority":"mid","parent":"Root Task Three","toggle":false}],
            newTask: "",
            newPriority: "high",
            newParent: "",
            addOpen: true
        }
    }
});

describe('ListContainer', () => {
  test('is a Vue instance', () => {
    expect(noPropWrapper.isVueInstance()).toBeTruthy()
  })
  test('renders correctly', () => {
    expect(noPropWrapper.element).toMatchSnapshot()
  })
  it("renders correct list title", () => {
    expect(titledWrapper.find('h1').text()).toEqual(title)
  })
  it("computes rootTasks correctly", () => {
    expect(taskedWrapper.vm.rootTasks).toEqual([{"title":"Root Task One","completed":false,"priority":"mid","parent":"","toggle":false},{"title":"Root Task Two","completed":false,"priority":"high","parent":"","toggle":false},{"title":"Root Task Three","completed":false,"priority":"low","parent":"","toggle":false},{"title":"Root Task Four","completed":false,"priority":"mid","parent":"","toggle":false},{"title":"Root Task Five","completed":false,"priority":"low","parent":"","toggle":false},{"title":"Root Task Six","completed":false,"priority":"high","parent":"","toggle":false}])
  })
  it("computes childTasks correctly", () => {
    expect(taskedWrapper.vm.childTasks).toEqual([{"title":"Sub-task One","completed":true,"priority":"low","parent":"Root Task One","toggle":false},{"title":"Sub-task Two","completed":false,"priority":"low","parent":"Root Task One","toggle":false},{"title":"Sub-task Three","completed":false,"priority":"mid","parent":"Root Task Three","toggle":false}])
  })
})
