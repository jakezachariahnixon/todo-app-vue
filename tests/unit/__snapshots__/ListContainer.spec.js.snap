// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListContainer renders correctly 1`] = `
<div
  id="listContainer"
>
  <h1
    id="listName"
  >
    
  </h1>
   
  <ul
    id="listTasks"
  />
   
  <button
    id="addTaskBtn"
  >
    
    +
  
  </button>
   
  <div
    id="addTaskInterface"
  >
    <div
      id="closeAddTaskInterface"
    >
      
      Ã—
    
    </div>
     
    <textarea
      id="taskName"
      placeholder="What do you need to do?"
    />
     
    <div
      id="priorityContainer"
    >
      <h2>
        Priority
      </h2>
       
      <button
        class="priorityBtn"
        id="lowPriority"
      >
        
        Low
      
      </button>
       
      <button
        class="priorityBtn"
        id="midPriority"
      >
        
        Medium
      
      </button>
       
      <button
        class="priorityBtn"
        id="highPriority"
      >
        
        High
      
      </button>
    </div>
     
    <div
      id="parentContainer"
    >
      <h2>
        Parent
      </h2>
       
      <select>
        <option
          value=""
        >
          None
        </option>
         
      </select>
    </div>
  </div>
</div>
`;
